Program.cs

```diff
@@ -202,6 +204,43 @@ public static void Main(string[] args)
                     // repackage for distribution
                     runCommand("ditto", $"-ck --rsrc --keepParent --sequesterRsrc {stagingApp} {zippedApp}");
 
+                    break;
+                
+                case RuntimeInfo.Platform.Linux:
+                    // avoid use of unzip on Linux system, it is not preinstalled by default
+                    ZipFile.ExtractToDirectory("osu!.AppDir-template.zip", $"{stagingPath}");
+                    
+                    // mark AppRun as executable, as zip does not contains executable information
+                    runCommand("chmod", $"+x {stagingPath}/osu!.AppDir/AppRun");
+                    
+                    runCommand("dotnet", $"publish -f netcoreapp3.1 -r linux-x64 {ProjectName} -o {stagingPath}/osu!.AppDir/usr/bin/ --configuration Release /p:Version={version} --self-contained");
+                    
+                    // mark output as executable
+                    runCommand("chmod", $"+x {stagingPath}/osu!.AppDir/usr/bin/osu!");
+
+                    // copy png icon (for desktop file)
+                    File.Copy(Path.Combine(solutionPath, "assets/lazer.png"), $"{stagingPath}/osu!.AppDir/osu!.png");
+
+                    // download appimagetool
+                    using (var client = new WebClient())
+                        client.DownloadFile("https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage", $"{stagingPath}/appimagetool.AppImage");
+                    
+                    // mark appimagetool as executable
+                    runCommand("chmod", $"a+x {stagingPath}/appimagetool.AppImage");
+
+                    // create AppImage itself
+                    // gh-releases-zsync stands here for GitHub Releases ZSync, that is a way to check for updates
+                    // ppy|osu|latest stands for https://github.com/ppy/osu and get the latest release
+                    // osu-*x86_64.AppImage.zsync is wild card to get AppImage update information file, that is generated by the tool
+                    // more information there https://docs.appimage.org/packaging-guide/optional/updates.html?highlight=update#using-appimagetool
+                    runCommand($"{stagingPath}/appimagetool.AppImage", $"\"{stagingPath}/osu!.AppDir\" -u \"gh-releases-zsync|ppy|osu|latest|osu!-*x86_64.AppImage.zsync\" \"{Path.Combine(Environment.CurrentDirectory, "releases")}/osu!-x86_64.AppImage\"", false);

```

Comment:
I suggest checking if we can find the Linux GPG sign keys here and then run a unsigned build if none are found.

Reply:
Using `--sign` on AppImage will try to sign it if possible, so I fixed that